Cod3rsGrowth.Web/wwwroot/Webapp/app/vendas/ListagemVenda.controller.js

```diff
@@ -0,0 +1,155 @@
+sap.ui.define([
+    "ui5/carro/app/common/BaseController",
+    "sap/ui/model/json/JSONModel",
+    "ui5/carro/model/formatter",
+    "sap/m/MessageBox"
+
+], function (BaseController, JSONModel, Formatter, MessageBox) {
+    "use strict";
+
+    const id = "id";
+    const modeloVenda = "Vendas";
+    const idDoFiltroCpf = "FiltroCpf";
+    const rotaListagem = "appListagem";
+    const rotaDetalhes = "appDetalhes";
+    const idDoFiltroNome = "FiltroNome";
+    const quantidadeDeCaracteresDoCpf = 14;
+    const quantidadeDeCaracteresDoTelefone = 15;
+    const idDoFiltroTelefone = "FiltroTelefone";
+    let url = "http://localhost:5071/api/Vendas";    
+    const rotaListagemCarros = "appListagemCarro";
+    const rotaAdicionarVenda = "appAdicionarVenda";
+
+    return BaseController.extend("ui5.carro.app.vendas.ListagemVenda", {
+        formatter: Formatter,
+
+        onInit() {
+            this.aoCoincidirRota();
+        },
+
+        _carregarListaDeVendas() {
+            let sucesso = true;
+
+            fetch(url)
+                .then((res) => {
+                    if (!res.ok)
+                        sucesso = false;
+                    return res.json()
+                })
+                .then((data) => {
+                    const jsonModel = new JSONModel(data)
+
+                    sucesso ? this.getView().setModel(jsonModel, modeloVenda)
+                        : this._erroNaRequisicaoDaApi(data);
+                })
+                .catch((err) => MessageBox.error(err));
+        },
+
+        aoCoincidirRota() {
+            this.processarEvento(() => {
+                var rota = sap.ui.core.UIComponent.getRouterFor(this);
+                rota.getRoute(rotaListagem).attachMatched(this._carregarListaDeVendas, this);
+            });
+        },
+        
+        aoColetarNome() {
+            const nome = this.oView.byId(idDoFiltroNome).getValue();
+            return nome;
+        },
+
+        aoColetarCpf() {
+            const cpf = this.oView.byId(idDoFiltroCpf).getValue();
+            if (cpf.length < quantidadeDeCaracteresDoCpf) return '';
+            return cpf;
+        },
+
+        aoColetarTelefone() {
+            const telefone = this.oView.byId(idDoFiltroTelefone).getValue();
+            if (telefone.length < quantidadeDeCaracteresDoTelefone) return '';
+            return telefone;
+        },
+
+        aoColetarDataInicial(oEvent) {
+            let dataInicial = oEvent.getParameter("from");;
+            let dataInicialFormatada = Formatter.formatarData(dataInicial);
+            
+             return dataInicialFormatada;
+        },
+
+        aoColetarDataFinal(oEvent) {
+            let dataFinal = oEvent.getParameter("to");
```

Comment:
muda o nome e faz o retorno direto coloca esse "to" em uma constante dentro de metodo